backend/packages/Upgrade/src/api/controllers/ExperimentController.ts

```diff
@@ -836,15 +841,17 @@ export class ExperimentController {
    */
   @Get('/single/:id')
   @OnUndefined(ExperimentNotFoundError)
-  public one(@Param('id') id: string, @Req() request: AppRequest): Promise<ExperimentDTO> | undefined {
-    if (!isUUID(id)) {
-      return Promise.reject(
-        new Error(
-          JSON.stringify({ type: SERVER_ERROR.INCORRECT_PARAM_FORMAT, message: ' : id should be of type UUID.' })
-        )
-      );
+  public async one(
+    @Params({ validate: true }) { id }: IdValidator,
+    @Req() request: AppRequest
+  ): Promise<ExperimentDTO> {
+    const experiment = await this.experimentService.getSingleExperiment(id, request.logger);
+
+    if (!experiment) {
```

Comment:
You don't need this if statement and throw the exception. We already have OnUndefined decorator which will set the proper HTTP code if undefined is returned.